src/rpc/misc.cpp

```diff
@@ -37,6 +40,219 @@
 
 #include <boost/format.hpp>
 
+unsigned char hexval(unsigned char c)
+{
+    if ('0' <= c && c <= '9')
+        return c - '0';
+    else if ('a' <= c && c <= 'f')
+        return c - 'a' + 10;
+    else if ('A' <= c && c <= 'F')
+        return c - 'A' + 10;
+    else abort();
+}
+
+void hex2ascii(const std::string& in, std::string& out)
```

Comment:
Merit already has ParseHex function which returns a vector of chars. You can take that and create a string from it. Please remove hex2ascii.